File: Domain/TransApp.Domain/Shipment/ShipmentDetailModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace TransApp.Domain.Shipment
{
    public class ShipmentDetailModel 
    {
        public int Id { get; set; }

        public int? ShipmentId { get; set; }

        public decimal? Quantity { get; set; }

        public int? PackTypeId { get; set; }

        public int? Length { get; set; }

        public int? Width { get; set; }

        public int? Height { get; set; }

        public int? Weight { get; set; }

        public int? ParentDetailId { get; set; }

        public int? UserIdCreated { get; set; }

        public DateTime? DateCreated { get; set; }

        public int? UserIdModified { get; set; }

        public DateTime? DateModified { get; set; }

    }

}

File: Site/Presentation/TransApp.Site.FrontEnd/src/app/shipment/models/shipment-detail-model.ts
export class ShipmentDetailModel {
  public id: number;
  public shipmentId: number;
  public quantity: number;
  public packTypeId: number;
  public length: number;
  public width: number;
  public height: number;
  public weight: number;
  public toRemove: boolean;

  public quantityDetail: number;
  public packTypeDetailId: number;

  // property used to store value of new created packtype
  public addedPackType: string;

  public userIdCreated: number;
  public dateCreated: Date;
  public userIdModified: number;
  public dateModified: Date;
}
